tests/test_lesion.py

```diff
@@ -0,0 +1,215 @@
+
+
+    expected_transfom = [ActivationsD, AsDiscreteD, FillHolesD, RemoveSmallObjectsD]
+
+    for t, e in zip(output_transform.transforms, expected_transfom):
+        assert isinstance(t, e)
+
+
+def test_predict_2dto3d(lesion_obj):
+    """
+    Tests the prediction function for lesion object.
+    verifies the functionality by performing predictions on a 3D volume (with size 64,64,...).
+
+    Parameters:
+    ----------
+    lesion_obj (object): The lesion object that provides the predict_2dto3d function.
+    ----------
+    """
+
+    liver_mask = segment_liver(
+        prediction_path=config.test["test_volume"],
+        modality="CT",
+        inference="sliding_window",
+        cp_path=config.test["reference_sliding_window"],
+    )
+
+    volume_dir = config.test["test_volume"]
+    close = MorphologicalClosing(iters=4)
+    prediction = lesion_obj.predict_2dto3d(
+        volume_path=volume_dir,
+        liver_mask=close(liver_mask[0]).permute(3, 0, 1, 2)
+
+    )
+
+
+    assert isinstance(prediction, torch.Tensor)
+    assert prediction.shape[0] == 1
+    assert prediction.shape[1] == 1  # number of channels
+
+    assert (
+        prediction.shape[2:] == nib.load(config.test["test_volume"]).get_fdata().shape
+    )
+    assert torch.min(prediction).item() == 0
+    assert torch.max(prediction).item() == 1
+
+def test_segment_lesion():
+    """
+    Tests segment_lesion function (lesion_inference = '3D').
+    verifies the functionality by performing segmentation using a 3D volume (with size 64,64,...)
+    Compares the results with the true reference segmentation.
+    """
+    lesion_prediction = segment_lesion(
+        prediction_path=config.test["test_volume"],
+        liver_inference="sliding_window",
+        lesion_inference="3D",
+        liver_cp=config.test["reference_sliding_window"],
+        lesion_cp=config.test["reference_lesion_cp"],
+    )
+
+    # Assertion checks for lesion_prediction
+    assert isinstance(lesion_prediction, torch.Tensor)
+    assert lesion_prediction.shape[0] == 1
+    assert lesion_prediction.shape[1] == 1
+    assert (
+        lesion_prediction.shape[2:]
+        == nib.load(config.test["test_volume"]).get_fdata().shape
+    )  # original volume dimension
+
+    lesion_prediction = lesion_prediction.cpu()
+    lesion_prediction = lesion_prediction.numpy()
+
+
+    ref_path = "tests/testdata/testcases/predicted_array_lesion.npy"
+    true = np.load(ref_path)
+    assert np.allclose(lesion_prediction, true, 0.01)
+
+
+def test_train():
+    """
+    Tests train function (epoch: 1 , inference = '3D', volume = '64,64,..' ) .
+    Compares the weights of the resulted checkpoint with the reference checkpoint's weights.
+    """
+    model = LesionSegmentation(inference="3D")
+    # load reference checkpoint of training 1 epoch on a volume
+    model.load_checkpoint(config.test["reference_lesion_cp"])
+    reference_weights = model.network.state_dict()
```

Comment:
I think it will be more understandable if you move these two lines (loading reference cp) below loading the training cp, because it seems as if 'model' is what is being trained. This is also a suggestion.